server/SwipeDecision.py

```diff
@@ -15,16 +16,66 @@ def getPotentialMatchList(currentUserId):
             cursor.execute(PotentialMatchQuery, userID)
             potentialListId = [i[0] for i in cursor.fetchall()]
 
+            newPotentialMatchList = []
+
+            if(len(potentialListId) != 0):
+
+                findLocation = "SELECT Users.longitude, Users.latitude, Profile.maxDistance FROM Users INNER JOIN Profile ON Users.id=Profile.userId WHERE Users.id = %s"
+                cursor.execute(findLocation, userID)
+                currentUserLocation = cursor.fetchone()
+                currentUserLongitude = readLocation(currentUserLocation)[0]
+                currentUserLatitude = readLocation(currentUserLocation)[1]
+                currentUserMaxDistance = readLocation(currentUserLocation)[2]
+
+                if(currentUserLongitude != None):
+
+                    for pmatch in range(len(potentialListId)):
+                        ID = (potentialListId[pmatch],)
+                        cursor.execute(findLocation, ID)
+                        potentialMatchLocation = cursor.fetchone() 
+                        potentialMatchLongitude = readLocation(potentialMatchLocation)[0]
+                        potentialMatchLatitude = readLocation(potentialMatchLocation)[1]
+                        potentialMatchMaxDistance = readLocation(potentialMatchLocation)[2]
+
+                        if(potentialMatchLongitude != None):
+                            distance = calculateDistance(currentUserLongitude, currentUserLatitude, potentialMatchLongitude, potentialMatchLatitude)
+
+                            if ((distance <= currentUserMaxDistance) & (distance <= potentialMatchMaxDistance)):
```

Comment:
you need two &. should be `((distance <= currentUserMaxDistance) && (distance <= potentialMatchMaxDistance))`